### PR TITLE
Upgrade react sdk to v8

### PR DESCRIPTION
**Pull request by andreiborza**
Closes: #530

### CODE REVIEW

File: react/src/components/About.jsx [hunk 1]
```diff
@@ -41,9 +41,7 @@ function About({ backend }) {
     // Error Handling
     response.forEach((r) => {
       if (!r.ok) {
-        Sentry.configureScope(function (scope) {
-          Sentry.setContext('response', r);
-        });
+        Sentry.getCurrentScope().setContext('response', r);
```
Comment (mydea) [inline react/src/components/About.jsx:44]: let's instead make this:

```js
Sentry.withScope((scope) => {
  scope.setContext(...);
  Sentry.captureException(...);
});
```

This is the preferred pattern over using current scope like this!

File: react/src/components/Products.jsx [hunk 2]
```diff
@@ -103,11 +103,9 @@ function Products({ frontendSlowdown, backend }) {
       })
         .then((result) => {
           if (!result.ok) {
-            Sentry.configureScope(function (scope) {
-              Sentry.setContext('err', {
-                status: result.status,
-                statusText: result.statusText,
-              });
+            Sentry.getCurrentScope().setContext('err', {
```
Comment (mydea) [inline react/src/components/Products.jsx:106]: let's use `Sentry.setContext(..)` instead

File: react/src/components/ProductsJoin.jsx [hunk 3]
```diff
@@ -16,11 +16,9 @@ function ProductsJoin({ backend }) {
     })
       .then((result) => {
         if (!result.ok) {
-          Sentry.configureScope(function (scope) {
-            Sentry.setContext('err', {
-              status: result.status,
-              statusText: result.statusText,
-            });
+          Sentry.getCurrentScope().setContext('err', {
```
Comment (mydea) [inline react/src/components/ProductsJoin.jsx:19]: let's use `Sentry.setContext(..)` instead

File: react/src/index.js [hunk 4]
```diff
@@ -72,6 +72,8 @@ Sentry.init({
   release: RELEASE,
   environment: ENVIRONMENT,
   tracesSampleRate: 1.0,
+  tracingOrigins: tracingOrigins,
```
Comment (mydea) [inline react/src/index.js:75]:
```suggestion
```

File: react/src/index.js [hunk 5]
```diff
@@ -192,86 +165,85 @@ class App extends Component {
...
-        RAGECLICK = true;
-      }
+    const scope = Sentry.getCurrentScope();
+    const customerType = [
+      'medium-plan',
+      'large-plan',
+      'small-plan',
+      'enterprise',
+    ][Math.floor(Math.random() * 4)];
+    scope.setTag('customerType', customerType);
```
Comment (mydea) [inline react/src/index.js:175]: let's not use current scope here, and just do `Sentry.setTag(...)` etc!

**Review by mydea — COMMENTED** [review]
generally looks good, mostly I'd replace usage of `getCurrentScope()` with just using top level methods!

File: react/src/components/Products.jsx (hunk 2 above)
Comment (andreiborza) [reply react/src/components/Products.jsx]: Thanks, updated.

File: react/src/components/ProductsJoin.jsx (hunk 3 above)
Comment (andreiborza) [reply react/src/components/ProductsJoin.jsx]: Updated.

File: react/src/index.js (hunk 5 above)
Comment (andreiborza) [reply react/src/index.js]: Updated and here and the other places that used this scope.

File: react/package.json [hunk 6]
```diff
@@ -19,7 +19,7 @@
     "web-vitals": "~1.0.1"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "react-app-rewired start",
```
Comment (andreiborza) [inline react/package.json:22]: This breaks HMR, see https://github.com/sentry-demos/empower/pull/381

Need to rewire jsx-runtime but also want HMR... 🤔

Comment (andreiborza) [reply react/package.json]: HMR was already broken, but I reverted this change anyway...

**Review by mydea — APPROVED** [review]
LGTM! 🚀

File: react/src/index.js [hunk 7]
```diff
@@ -80,44 +81,15 @@ Sentry.init({
...
-      tracingOrigins: tracingOrigins,
-      tracePropagationTargets: tracingOrigins,
-      routingInstrumentation: Sentry.reactRouterV6Instrumentation(
-        useEffect,
-        useLocation,
-        useNavigationType,
-        createRoutesFromChildren,
-        matchRoutes
-      ),
-      beforeNavigate: (context) => {
```
Comment (cstavitsky) [inline react/src/index.js:95]: @andreiborza to confirm, is the reason for removing the beforeNavigate block that it's outdated and no longer necessary?

Comment (andreiborza) [reply react/src/index.js:95]: yes, the newer `reactRouterV6BrowserTracingIntegration` doesn't have an option for that and from what I can tell, it's (just?) doing route parametrization which the newer integration does out of the box.

Could you also chime in here please @mydea?

**Review by cstavitsky — APPROVED** [review]

File: react/src/index.js (hunk 7 above)
Comment (mydea) [reply react/src/index.js:95]: sorry, saw this too late! This should not be needed here, if needed you can use `beforeStartSpan` instead, see https://docs.sentry.io/platforms/javascript/tracing/instrumentation/automatic-instrumentation/#beforestartspan

**Pull request merged**